adagrams/game.py

```diff
@@ -1,11 +1,163 @@
-def draw_letters():
-    pass
+import random
+from operator import itemgetter
 
-def uses_available_letters(word, letter_bank):
-    pass
+def build_letter_pool(): # wave 1
+    
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+    }
+
+    letter_pool = []
+
+    for letter, frequency in LETTER_POOL.items():
+        for num in range(frequency):
+            letter_pool.append(letter)
+    # print(letter_pool)
+    return letter_pool
+    
+
+def draw_letters(): #wave 1
+    letter_pool = build_letter_pool()
+    letters = random.sample(letter_pool, 10)
+    # print(letters)
+    return letters
+    # pass
+
+def uses_available_letters(word, letter_bank): #wave 2
+    result = True
+    letter_bank_dict = {}
+    
+    for letter in letter_bank: 
+        letter_bank_dict[letter] = letter_bank.count(letter)
+    
+    word = word.upper()
+    
+    for character in word: 
+        if character in letter_bank_dict:
+            if letter_bank_dict[character] == 0:
+                result = False
+            letter_bank_dict[character] -= 1
+        else:
+            result = False
+    return result
+    # pass
+
+
+#-----wave 03 ------------ #
+     #LP's version (with coworking!)
 
 def score_word(word):
-    pass
+    letter_scores = make_score_dict()
+    score = get_score(word.upper(), letter_scores)
+    return score
+    
+#challenge: compare words to an imported dictionary to see if it's a word!
```

Comment:
Cool idea! Perhaps you can explore the [english-words](https://pypi.org/project/english-words/) or [PyDictionary](https://pypi.org/project/PyDictionary/) modules